### PR TITLE
[rush] Run repo analysis after initial operation creation.

### PR DESCRIPTION
**Pull request by iclanton**
## Summary

Refactor `PhasedScriptAction` to run repo analysis after initial operation creation.

This removes the `projectChangeAnalyzer` property from the context passed to `createOperations`.

## How it was tested

Ran `rush build` with the change in the rushstack repo and observed that projects came from cache.

## Impacted documentation

API docs will need to be updated.

### CODE REVIEW

File: libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 1]
```diff
@@ -660,7 +689,10 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
       );
 
       const executeOptions: IExecutionOperationsOptions = {
-        createOperationsContext,
+        executeOperationsContext: {
+          ...initialCreateOperationsContext,
+          projectChangeAnalyzer: initialState
+        },
```
Comment (dmichon-msft) [inline libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:695]:
```suggestion
        executeOperationsContext: createOperationsContext
```

Why are you overriding `projectChangeAnalyzer` here?

Comment (iclanton) [reply libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:695]: It's removed from `createOperationsContext` in this PR.

File: libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 2]
```diff
@@ -484,16 +480,43 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
     }
   }
 
-  private async _runInitialPhases(options: IRunPhasesOptions): Promise<void> {
-    const { initialCreateOperationsContext, executionManagerOptions, stopwatch, terminal } = options;
+  private async _runInitialPhases(options: IInitialRunPhasesOptions): Promise<IRunPhasesOptions> {
+    const {
+      initialCreateOperationsContext,
+      executionManagerOptions: partialExecuteionManagerOptions,
```
Comment (dmichon-msft) [inline libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:486]:
```suggestion
      executionManagerOptions: partialExecutionManagerOptions,
```

**Review by dmichon-msft — APPROVED** [review]
Typo, otherwise looks fine.

**Pull request merged**